apps/openmw/mwmechanics/aicombat.cpp

```diff
@@ -490,6 +490,28 @@ namespace MWMechanics
 
     void AiCombatStorage::startCombatMove(bool isDistantCombat, float distToTarget, float rangeAttack, const MWWorld::Ptr& actor, const MWWorld::Ptr& target)
     {
+        // get the range of the target's weapon
+        float rangeAttackOfTarget = 0.f;
+        MWWorld::Ptr targetWeapon = MWWorld::Ptr();
+        const MWWorld::Class& targetClass = target.getClass();
+
+        if (targetClass.hasInventoryStore(target))
+        {
+            MWMechanics::WeaponType weapType = WeapType_None;
+            MWWorld::ContainerStoreIterator weaponSlot =
+                MWMechanics::getActiveWeapon(targetClass.getCreatureStats(target), targetClass.getInventoryStore(target), &weapType);
+            if (weapType != WeapType_PickProbe && weapType != WeapType_Spell && weapType != WeapType_None && weapType != WeapType_HandToHand)
+                targetWeapon = *weaponSlot;
+        }
+
+        std::shared_ptr<Action> targetWeaponAction(new ActionWeapon(targetWeapon));
+
+        bool isRangedCombat = false;
+        if (targetWeaponAction.get())
+        {
+            rangeAttackOfTarget = targetWeaponAction->getCombatRange(isRangedCombat);
+        }
```

Comment:
Something seems odd here. If `targetWeaponAction` is allocated with `new`, won't it always be a non-null pointer and thus `targetWeaponAction.get()` always evaluate to `true`? Or am I missing something?
And if it doesn't need to be shared with anything, maybe it should be changed to a `unique_ptr` instead?